source/common/http/utility.cc

```diff
@@ -609,7 +614,16 @@ void Utility::sendLocalReply(const bool& is_reset, const EncodeFunctions& encode
           body_text[body_text.length() - 1] == '\n') {
         body_text = body_text.substr(0, body_text.length() - 1);
       }
-      response_headers->setGrpcMessage(PercentEncoding::encode(body_text));
+      auto encoded_body_text = PercentEncoding::encode(body_text);
+      response_headers->setGrpcMessage(encoded_body_text);
+      if (local_reply_data.grpc_error_details_) {
+        local_reply_data.grpc_error_details_->set_message(encoded_body_text);
+      }
+    }
+    if (local_reply_data.grpc_error_details_) {
+      auto status_detail_str = local_reply_data.grpc_error_details_->SerializeAsString();
```

Comment:
nit: too much use of `auto`. Only use auto if type is obvious (iterators, return value of `std::make_unique`, etc.)

Would be better to declare this as `std::string status_detail`